src/storage/quota/QuotaStrategy.ts

```diff
@@ -0,0 +1,46 @@
+import type { PassThrough, Readable } from 'stream';
+import type { RepresentationMetadata } from '../../http/representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../../http/representation/ResourceIdentifier';
+import type { Guarded } from '../../util/GuardedStream';
+import type { Size } from '../size-reporter/Size';
+
+/**
+ * A QuotaStrategy is used when we want to set a limit to the amount of data that can be
+ * stored on the server.
+ * This can range from a limit for the whole server to a limit on a per pod basis.
+ * The way the size of a resource is calculated is implemented by the implementing classes.
+ * This can be bytes, quads, file count, ...
+ */
+export interface QuotaStrategy {
+  /**
+   * Get the available space given a resource's identifier.
+   * If the given resource already exists it will deduct the already taken up
+   * space by that resource since it is going to be overwritten and thus counts
+   * as available space.
+   *
+   * @param identifier - the identifier of the resource of which you want the available space
+   * @returns the available space and the unit of the space as a Size object
+   */
+  getAvailableSpace: (identifier: ResourceIdentifier) => Promise<Size>;
+
+  /**
+   * Get an estimated size of the resource
```

Comment:
At the moment when no valuable estimation can be made it will return `0` as amount. At least that is how I implemented it for the current two strategies. If the estimated size is only used in relation to quota I would prefer this.

If you agree, I should idd add a comment explaining this in the code. (test cases already exist)
If you don't agree, it requires only a few lines of the implementation to be changed.

Lmk.

Reply:
`undefined` more aptly indicates the lack of estimation, where `0` could be a real informative estimation of size zero.